controllers/common/common.go

```diff
@@ -60,24 +64,24 @@ func (r *Client) UpdateMeta(ctx context.Context, object client.Object) {
 }
 
 //UpdateStatus function updates the status based on the process step
-//func (r *Client) UpdateStatus(ctx context.Context, obj runtime.Object, state managerv1alpha1.State, requeueTime ...float64) (ctrl.Result, error) {
-//	log := log.Logger(ctx, "controllers.common", "common", "UpdateStatus")
-//
-//	if err := r.Status().Update(ctx, obj); err != nil {
-//		log.Error(err, "Unable to update status", "status", state)
-//		r.Recorder.Event(obj, v1.EventTypeWarning, string(managerv1alpha1.Error), "Unable to create/update status due to error "+err.Error())
-//		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-//	}
-//
-//	if state != managerv1alpha1.Error {
-//		return ctrl.Result{}, nil
-//	}
-//
-//	//if wait time is specified, requeue it after provided time
-//	if len(requeueTime) == 0 {
-//		requeueTime[0] = 0
-//	}
-//
-//	log.Info("Requeue time", "time", requeueTime[0])
-//	return ctrl.Result{RequeueAfter: time.Duration(requeueTime[0]) * time.Millisecond}, nil
-//}
+func (r *Client) UpdateStatus(ctx context.Context, obj client.Object, state alertmanagerv1alpha1.State, requeueTime ...float64) (ctrl.Result, error) {
+	log := log.Logger(ctx, "controllers.common", "common", "UpdateStatus")
+
+	if err := r.Status().Update(ctx, obj); err != nil {
+		log.Error(err, "Unable to update status", "status", state)
+		r.Recorder.Event(obj, v1.EventTypeWarning, string(alertmanagerv1alpha1.Error), "Unable to create/update status due to error "+err.Error())
```

Comment:
how does the event record handle different event type?

Reply:
It doesn't have to handle anything different. K8s events are of 2 types - 1. Normal 2. Warning- When you do 
```k8s get events``` it needs that type so we are passing what type of event that is as part of event creation

```
mesh-health-check                     5m49s       Normal    SawCompletedJob           cronjob/mesh-health-check-cronjob                                (combined from similar events): Saw completed job: mesh-health-check-cronjob-1624644900, status: Complete
mesh-health-check                     11m         Warning   Unhealthy                 pod/mesh-health-check-deployment-f4b989b88-t7rjh                 Readiness probe failed: Get http://10.192.20.50:15021/healthz/ready: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```